subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/espresso/action/TouchOutsideAction.kt

```diff
@@ -0,0 +1,88 @@
+package com.avito.android.lib.actions
+
+import android.graphics.Rect
+import android.os.SystemClock
+import android.view.InputDevice
+import android.view.MotionEvent
+import android.view.MotionEvent.PointerProperties
+import android.view.View
+import android.view.ViewConfiguration
+import android.view.ViewGroup
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.CoordinatesProvider
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.Configurator
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.Matcher
+
+class TouchOutsideAction : ViewAction {
+
+    override fun getConstraints(): Matcher<View> = ViewMatchers.isAssignableFrom(ViewGroup::class.java)
+
+    override fun getDescription(): String = "Press outside of dialog"
+
+    override fun perform(uiController: UiController, view: View) {
+        val coordinate = OutSideCoordinateProvider().calculateCoordinates(view)
+
+        val x = coordinate[0]
+        val y = coordinate[1]
+
+        val properties = PointerProperties()
+        properties.id = 0
+        properties.toolType = Configurator.getInstance().toolType
+
+        val coords = MotionEvent.PointerCoords()
+        coords.pressure = 1f
+        coords.size = 1f
+        coords.x = x
+        coords.y = y
+
+        val cur = SystemClock.uptimeMillis()
+        val events = listOf(
+            MotionEvent.obtain(
```

Comment:
extract to function and give a comment on why you configure this event with these params, please

Reply:
done